sdk/include/opentelemetry/sdk/metrics/export/metric_producer.h

```diff
@@ -31,14 +32,40 @@ namespace metrics
  */
 struct ScopeMetrics
 {
-  const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope_;
+  const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope_ = nullptr;
   std::vector<MetricData> metric_data_;
+
+  template <class ScopePtr, class MetricDataType>
+  inline ScopeMetrics(ScopePtr &&scope, MetricDataType &&metric)
+      : scope_{std::move(scope)}, metric_data_{std::move(metric)}
+  {}
+
+  inline ScopeMetrics()                     = default;
+  inline ScopeMetrics(const ScopeMetrics &) = default;
+  inline ScopeMetrics(ScopeMetrics &&)      = default;
+
+  inline ScopeMetrics &operator=(const ScopeMetrics &) = default;
+
+  inline ScopeMetrics &operator=(ScopeMetrics &&) = default;
```

Comment:
just curious, why is this stuff necessary?

Reply:
Some old versions of GCC have BUGs with default member initializer.Please see the details here: https://github.com/grpc/grpc/issues/28147

Reply:
Ack, thanks.

FTR, I am not going to review this PR. I stumbled upon the PR randomly and left questions, because I was curious. Thanks for answering them!